File: UI/QACheckSheetUI/src/features/ngDetail/types/ngDetail.ts
export interface NgDetail {
    resultId: number | null;
    ngId: number | null;
    sheetCode?: string;
    sheetName?: string;
    deviceCode?: string;
    deviceName?: string;
    pathTitles?: string;
    dataType?: string;
    value?: string;
    status?: string;
    min?: number | null;
    max?: number | null;
    checkedBy?: string;
    checkedDate?: Date | string | null;
    ngContentDetail?: string;
    fixContent?: string;
    fixedBy?: string;
    fixedDate?: Date | string | null;
    confirmedBy?: string;
    confirmedDate: Date | string | null;
    note?: string;
}

export interface UpdateValueResultPayload {
    resultId: number;
    value?: string;
    status?: string;
}

export interface ConfirmNgPayload {
    ngId: number;
    resultId: number;
    confirmedBy: string;
    confirmedDate: Date | string;
}

File: API/QACheckSheetAPI/QACheckSheetAPI/Models/Domain/NGDetail.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace QACheckSheetAPI.Models.Domain
{
    public class NGDetail
    {
        [Key]
        public int NgId { get; set; }
        [Required]
        public long ResultId { get; set; }
        [MaxLength(700)]
        public string NGContentDetail { get; set; } = string.Empty; // Chi tiết nội dung NG
        [MaxLength(700)]
        public string FixContent { get; set; } = string.Empty; // Nội dung khắc phục
        public string FixedBy { get; set; } = string.Empty;
        public DateTime? FixedDate { get; set; }
        public string ConfirmedBy { get; set; } = string.Empty;
        public DateTime? ConfirmedDate { get; set; }
        public string Note { get; set; } = string.Empty;

        //Navigation
        [ForeignKey("ResultId")]
        public CheckResult? CheckResult { get; set; }
    }
}
